mesonbuild/interpreter/compiler.py

```diff
@@ -96,6 +96,7 @@ class PreprocessKW(TypedDict):
         compile_args: T.List[str]
         include_directories: T.List[build.IncludeDirs]
         dependencies: T.List[dependencies.Dependency]
+        depends: T.List[T.Union[build.BuildTarget, build.CustomTarget]]
```

Comment:
We should also support `build.CustomTargetIndex` here, which would allow things like `preprocess(..., depends : ct[1])`